fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IDatabaseTranslator.java

```diff
@@ -214,4 +215,12 @@
      * @return
      */
     default boolean isIndexUseSchemaPrefix() { return true; }
+
+    /**
+     * @return the total number of parameters supported by the data store during a query.
+     *  An empty optional indicates no maximum is set.
+     */
+    default Optional<Integer> maximumQueryParameters() {
+        return Optional.empty();
+    }
```

Comment:
I chose to add this as we hit issues, we can implement for other providers.  My guess is derby/db2 have a very high parameter count.